Sources/XcodeGraph/Models/Project.swift

```diff
@@ -55,8 +55,8 @@ public struct Project: Hashable, Equatable, CustomStringConvertible, CustomDebug
     /// The version in which a check happened related to recommended settings after updating Xcode.
     public var lastUpgradeCheck: Version?
 
-    /// Indicates whether the project is imported through `Package.swift`.
-    public var isExternal: Bool
+    /// A type that indicates where the project is coming from.
+    public var type: ProjectType
```

Comment:
I'd advise leaving this and marking it as deprecated using the Swift annotations. Instead, I recommend moving this property down at the target level, and name it `type: TargetType`. The reason why I suggest moving it to the target level is because, down the road, we might build a tool for React Native where we need the flexibility to set this information at the target level.

Reply:
Also... by looking at the cases, `remotePackage`, `localPackage`, `tuistProject`, it's hard to tell what the properties are for and they are coupled to Tuist, which is something we want to avoid in the design of this package. What about coming up with a type and a property that reflects what the use cases are from the generation standpoint.
You know the use cases better than me so you might be able to provide better suggestions than me.

Reply:
Additionally, I've never worked with React Native before, so if there's anything else I should be aware of when moving this property to the target level, I'd appreciate it if you could let me know.

Reply:
Here's an idea. What if we extend target with:
```swift
let target = Target(tags: [.remote])
```

Where a tag is:

```swift
enum TargetTag {
  case remote
}
```

Having tags is something that we've been pondering for some time for other features, and I think it'd also solve the need that you have here.